spec/models/tour_spec.rb

```diff
@@ -0,0 +1,108 @@
+require 'rails_helper'
+
+RSpec.describe Tour, type: :model do
+  let(:category) {
+    Category.create!(
+      name: "Example category"
+    )
+  }
+
+  let(:@tour_1){
+    Tour.create!(
+      name: "Name Tour1 ",
+      description: "abc",
+      start_date: Time.zone.parse('2022-09-11 21:00'),
+      end_date: Time.zone.parse('2022-10-11 21:00'),
+      category_id: category.id,
+      price: 1,
+      stock: 1,
+      avg_rating: 1
+    )
+  }
+
+  let(:@tour_2){
+    Tour.create!(
+      name: "Name Tour2 ",
+      description: "abc",
+      start_date: Time.zone.parse('2022-10-11 21:00'),
+      end_date: Time.zone.parse('2022-11-11 21:00'),
+      category_id: category.id,
+      price: 2,
+      stock: 1,
+      avg_rating: 1
+    )
+  }
+
+  describe "presence" do
+    it { should validate_presence_of(:name)}
+    it { should validate_presence_of(:description)}
+    it { should validate_presence_of(:start_date)}
+    it { should validate_presence_of(:price)}
+    it { should validate_presence_of(:stock)}
+    it { should validate_presence_of(:avg_rating)}
+  end
+
+  describe "check scope" do
+    it "orders by id ascending" do
+      expect(Tour.incre_order).to eq([@tour_1, @tour_2])
+    end
+
+    it "order_rating" do
+      expect(Tour.order_rating(1)).to eq([@tour_1, @tour_2])
+    end
+
+    it "by start date" do
+      expect(Tour.by_start_date(Time.zone.parse('2022-10-11 21:00'))).to eq([@tour_2])
+    end
+
+    it "by end date" do
+      expect(Tour.by_end_date(Time.zone.parse('2022-11-11 21:00'))).to eq([@tour_1,@tour_2])
+    end
+
+    it "by most name" do
+      expect(Tour.by_most_name("Tour2")).to eq([@tour_2])
+    end
+
+    it "by_rating_array" do
+      expect(Tour.by_rating_array(1)).to eq([@tour_1, @tour_2])
+    end
+
+    it "by_start_price" do
+      expect(Tour.by_start_price(1)).to eq([@tour_1, @tour_2])
+    end
+
+    it "by_end_price" do
+      expect(Tour.by_end_price(1)).to eq([@tour_1])
+    end
+
+    it "by avg rating" do
+      expect(Tour.by_avg_ratings(1)).to eq([@tour_1, @tour_2])
+    end
+
+    context "order by name" do
+      it "ascending" do
+        expect(Tour.order_by_name("asc")).to eq([@tour_1, @tour_2])
+      end
+
+      it "descending" do
+        expect(Tour.order_by_name("desc")).to eq([@tour_2, @tour_1])
+      end
+    end
+
+    context "order by price" do
```

Comment:
Ý nghĩa của describer, context, it

Reply:
describe nhóm các test case theo model, controller, method, class 
context dùng để nhóm các điều kiện của example
it định nghĩa 1 test case